core/mem.c

```diff
@@ -0,0 +1,22 @@
+#include "mem.h"
+#include <stdlib.h>
+#include <string.h>
+
+struct ccord_memory_fn_vtable ccord_global_memory_fn_vtable;
+
+bool ccord_mem_is_initialized = false;
+
+void
+mem_init(malloc_fn_t malloc_fn,
+         calloc_fn_t calloc_fn,
+         realloc_fn_t realloc_fn,
+         free_fn_t free_fn,
+         strdup_fn_t strdup_fn)
+{
+    ccord_mem_is_initialized = true;
```

Comment:
Would it make sense to check if `ccord_mem_is_initialized` is already set here and emit an error if so? Or would there be a scenario where this function may be called more than once?

Reply:
You're correct. Changing memory functions on the fly does not make sense. I will add a check.